python_modules/libraries/dagster-docker/dagster_docker/external_resource.py

```diff
@@ -14,36 +13,50 @@
     ExtResource,
 )
 from dagster._core.ext.utils import (
-    ExtFileContextInjector,
-    ExtFileMessageReader,
+    ExtEnvContextInjector,
     io_params_as_env_vars,
 )
 from dagster_ext import (
     DagsterExtError,
+    ExtDefaultMessageWriter,
     ExtExtras,
+    ExtParams,
 )
 from pydantic import Field
-from typing_extensions import TypeAlias
 
-VolumeMapping: TypeAlias = Mapping[str, Mapping[str, str]]
 
+class DockerLogsMessageReader(ExtMessageReader):
+    @contextmanager
+    def read_messages(
+        self,
+        _context: ExtOrchestrationContext,
+    ) -> Iterator[ExtParams]:
+        yield {ExtDefaultMessageWriter.STDIO_KEY: ExtDefaultMessageWriter.STDERR}
 
-_CONTEXT_SOURCE_FILENAME = "context"
-_MESSAGE_SINK_FILENAME = "messages"
+    def consume_docker_logs(self, container, ext_context):
+        for log_line in container.logs(stdout=True, stderr=True, stream=True, follow=True):
+            try:
+                message = json.loads(log_line)
+                # need better message check
+                if message.keys() == {"method", "params"}:
+                    ext_context.handle_message(message)
```

Comment:
I think we should emit some sort of "reserved key" as part of an ext-emitted json object to tighten this up.

```
{
   "__dagster_ext_message" : true,
   # ... rest of the stuff
}
```

And does this work if the json contains strings with newlines? Does it properly do some sort of escaping?